x-pack/plugins/maps/public/classes/layers/vector_layer/mvt_vector_layer/mvt_vector_layer.tsx

```diff
@@ -124,21 +129,35 @@ export class MvtVectorLayer extends AbstractVectorLayer {
       }
     });
 
+    const canMultiCountShapes =
+      !this.getStyle().getIsPointsOnly() && totalFeaturesCount > 1 && tilesWithFeatures > 1;
+    const countPrefix = canMultiCountShapes ? '~' : '';
+    const countMsg = areResultsTrimmed
+      ? i18n.translate('xpack.maps.tiles.resultsTrimmedMsg', {
+          defaultMessage: `Results limited to {countPrefix}{count} documents.`,
+          values: {
+            count: totalFeaturesCount.toLocaleString(),
+            countPrefix,
+          },
+        })
+      : i18n.translate('xpack.maps.tiles.resultsCompleteMsg', {
+          defaultMessage: `Found {countPrefix}{count} documents.`,
+          values: {
+            count: totalFeaturesCount.toLocaleString(),
+            countPrefix,
+          },
+        });
+    const tooltipContent = canMultiCountShapes
+      ? countMsg +
+        i18n.translate('xpack.maps.tiles.shapeCountMsg', {
+          defaultMessage:
+            ' Documents may be counted multiple times if geometry crosses tile boundaries.',
```

Comment:
I like the messaging and overall approach. 

Not really sure how much we should wordsmith :) without falling in black hole, but going of @jsanz comments, something that hits that perfect balance between informative without giving away too much details.

trimmed: `Results limited to ~51123 documents. This count is approximate.`.
not-trimmed: `Found 426 documents. This count is approximate.`

?

Reply:
Both options look OK to me :+1: